src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```
@@ -0,0 +1,51 @@
+package mate.academy.service.impl;
+
+import java.util.Collections;
+import java.util.List;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        ShoppingCart cart = shoppingCartDao.getByUser(user).orElseThrow(() ->
+                new RuntimeException("Optional user by id: " + user.getId()
+                        + " is empty"));
+        Ticket ticket = ticketDao.add(new Ticket(movieSession, user));
+        List<Ticket> tickets = cart.getTickets();
+        tickets.add(ticket);
+        shoppingCartDao.update(cart);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        return shoppingCartDao.getByUser(user).get();
```

You can throw an exception here instead of addSession and re-use this method there.
Can also make the message more user-friendly: "User with id ... does not have a shopping cart".